src/interpret.jl

```diff
@@ -171,14 +171,18 @@ function evaluate_foreigncall(frame::Frame, call_expr::Expr)
     return Core.eval(moduleof(frame), Expr(head, args...))
 end
 
-# We have to intercept llvmcall before we try it as a builtin
+# We have to intercept ccalls / llvmcalls before we try it as a builtin
 function bypass_builtins(frame, call_expr, pc)
     if isassigned(frame.framecode.methodtables, pc)
         tme = frame.framecode.methodtables[pc]
         if isa(tme, Compiled)
             fargs = collect_args(frame, call_expr)
             f = to_function(fargs[1])
-            return Some{Any}(f(fargs[2:end]...))
+            if parentmodule(f) === JuliaInterpreter.CompiledCalls
+                return Some{Any}(Base.invokelatest(f, fargs[2:end]...))
```

Comment:
I wonder how much this hurts performance. Have you profiled? If the answer is "a lot," [this branch of SnoopCompile](https://github.com/timholy/SnoopCompile.jl/tree/teh/inference_timing) illustrates a dirty-rotten trick in its `__init__` function: we could theoretically set the world age to these constructed methods to be the same as the world age of the method we stole them from.

Reply:
How do I profile it when it errors without this? :P

Reply:
So by just calling things twice to get around the first world error I get

```
julia>  s = "foo"; @btime @interpret pointer_from_objref(s)
  3.737 μs (42 allocations: 2.34 KiB)
Ptr{Nothing} @0x0000000009463a10
```

so perhaps a 20% speedup?

Reply:
Not worth worrying about then, at least not until other bottlenecks go away.

Reply:
FWIW, on master:

```
julia> atomic_fence() = Base.llvmcall("""
                                 fence seq_cst
                                 ret void
                                 """, Cvoid, Tuple{})

julia> g() = atomic_fence()

julia> @interpret g()
┌ Warning: likely failure to return to toplevel, try JuliaInterpreter.split_expressions
└ @ JuliaInterpreter C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\interpret.jl:580
┌ Warning: likely failure to return to toplevel, try JuliaInterpreter.split_expressions
└ @ JuliaInterpreter C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\interpret.jl:580
ERROR: MethodError: no method matching llvmcall_83ebde91_bde5_4587_bba1_52f92157d766()
The applicable method may be too new: running in world age 25606, while current world is 25607.
```